ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java

```diff
@@ -358,44 +369,59 @@ public long size() {
     private final String key;
     private final Object value;
     private final long timestamp;
-    private final Window window;
+    private final WindowData window;
 
     Record(final Topic topic,
            final String key,
            final Object value,
            final long timestamp,
-           final Window window) {
+           final WindowData window) {
       this.topic = topic;
       this.key = key;
       this.value = value;
       this.timestamp = timestamp;
       this.window = window;
     }
 
+    private Serializer<?> keySerializer() {
+      final Serializer<String> stringDe = Serdes.String().serializer();
```

Comment:
perhaps we could have a `getSerde()` method that has the `if` in it and then use that in the two places here...